packages/valory/skills/termination_abci/tests/test_behaviours.py

```diff
@@ -0,0 +1,547 @@
+                state=State(
+                    ledger_id="ethereum",
+                    body=response_body,
+                ),
+            ),
+        )
+
+    def _mock_get_tx_data_request(
+        self,
+        error: bool = False,
+    ) -> None:
+        """Mock a MultiSendContract.get_tx_data() request."""
+        if not error:
+            response_performative = ContractApiMessage.Performative.RAW_TRANSACTION
+            response_body = dict(data=self._MOCK_TX_RESPONSE)
+        else:
+            response_body = dict()
+            response_performative = ContractApiMessage.Performative.ERROR
+        self.mock_contract_api_request(
+            contract_id=str(MultiSendContract.contract_id),
+            request_kwargs=dict(
+                performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+                contract_address=MULTISEND_ADDRESS,
+            ),
+            response_kwargs=dict(
+                performative=response_performative,
+                raw_transaction=RawTransaction(
+                    ledger_id="ethereum",
+                    body=response_body,
+                ),
+            ),
+        )
+
+    @pytest.mark.parametrize(
+        "test_case",
+        [
+            BehaviourTestCase(
+                name="agent fails to get the service owner",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[],
+                err_reqs=[_mock_get_service_owner_request],
+                expected_logs=[_SERVICE_OWNER_ERR_LOG],
+            ),
+            BehaviourTestCase(
+                name="agent fails to get zero transfer event",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[_mock_get_service_owner_request],
+                err_reqs=[_mock_get_zero_transfer_events_request],
+                expected_logs=[_ZERO_TRANSFER_EVENTS_ERR_LOG],
+            ),
+            BehaviourTestCase(
+                name="agent fails to get owner removal event",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[
+                    _mock_get_service_owner_request,
+                    _mock_get_zero_transfer_events_request,
+                ],
+                err_reqs=[_mock_get_removed_owner_events_request],
+                expected_logs=[_SERVICE_OWNER_REMOVED_EVENTS_ERR_LOG],
+            ),
+            BehaviourTestCase(
+                name="agent fails to get the safe owners",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[
+                    _mock_get_service_owner_request,
+                    _mock_get_zero_transfer_events_request,
+                    _mock_get_removed_owner_events_request,
+                ],
+                err_reqs=[_mock_get_owners_request],
+                expected_logs=[_SAFE_OWNERS_ERR_LOG],
+            ),
```

Comment:
I like this a lot, context managers on the text case, it's exactly what I wanted :+1: